staging/src/k8s.io/code-generator/cmd/conversion-gen/generators/conversion.go

```diff
@@ -303,10 +299,25 @@ func Packages(context *generator.Context, arguments *args.GeneratorArgs) generat
 			unsafeEquality = noEquality{}
 		}
 
+		path := pkg.Path
+		// if the source path is within a /vendor/ directory (for example,
+		// k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/apis/meta/v1), allow
+		// generation to output to the proper relative path (under vendor).
+		// Otherwise, the generator will create the file in the wrong location
+		// in the output directory.
+		// TODO: build a more fundamental concept in gengo for dealing with modifications
+		// to vendored packages.
+		if strings.HasPrefix(pkg.SourcePath, arguments.OutputBase) {
```

Comment:
this code is what we use elsewhere, e.g. in deepcopy-gen. The TODO is still valid, but at least this PR unifies different ugly hacks into one.